ibis/backends/base/sql/alchemy/__init__.py

```diff
@@ -206,31 +205,46 @@ def create_table(
             name, schema, database=database or self.current_database
         )
 
-        if has_expr := expr is not None:
-            # this has to happen outside the `begin` block, so that in-memory
-            # tables are visible inside the transaction created by it
-            self._register_in_memory_tables(expr)
+        if expr is None:
+            with self.begin() as bind:
+                table.create(bind=bind, checkfirst=force)
+            return
+
+        # this has to happen outside the `begin` block, so that in-memory
+        # tables are visible inside the transaction created by it
+        self._register_in_memory_tables(expr)
+
+        backends, _ = expr._find_backends()
+        if len(backends) == 1:
+            same_backend = backends[0] == self
+        elif len(backends) > 1:
+            raise ValueError('Multiple backends found')
+        else:
+            # Treat unbound exprs as based on this backend, matching `execute`
+            same_backend = True
```

Comment:
Is this logic something that should be extracted into the parent SQL backend class?